tools/generators/ethlike/contract_non_const_methods.go.tmpl

```diff
@@ -77,6 +77,16 @@ func ({{$contract.ShortVar}} *{{$contract.Class}}) {{$method.CapsName}}(
 		transaction,
 		transactorOptions,
 		func(newTransactorOptions *bind.TransactOpts) (*types.Transaction, error) {
+			// If original transactor options has a non-zero gas limit, that
+			// means the client code set it on their own. In that case, we
+			// should rewrite the gas limit from the original transaction
+			// for each resubmission. If the gas limit is not set by the client
+			// code, let the the submitter re-estimate the gas limit on each
+			// resubmission.
+			if transactorOptions.GasLimit != 0 {
+				newTransactorOptions.GasLimit = transaction.Gas()
```

Comment:
Why not just `= transactorOptions.GasLimit`? IMO it's much clearer than reading it from another struct.

Reply:
Yep, that's right. See: https://github.com/keep-network/keep-common/pull/84/commits/85c3f46bb66a4a679c2c1aad2d0f2a179ee2ef79